elizhahn/karaoke-stretch-api
# Seeding

## What does this PR do (summary of changes)?

 - Creates a seeding file for data
 - Installs and saves knex.js as a dependency
 - Utilizes knex migrations and seeding to set up localhost port 8080 as the local server with accurate database entries
 - Defines (replaces config pool) API endpoints for the songs and genres tables

## How should it be tested?

(Assuming the user has psql and an empty database named `carryokay_api`)

 - User should install the new dependencies by running the command `npm i` in the terminal at the head of the repository
 - User should set up the sessions migration with the command `knex migrate:latest`
 - User should process the seeding data with the command `knex seed:run`
 - User can open their server with either `nodemon` or `npm run start` on the main-server.js file to access the endpoints

## Future Iterations:

 - Eventually desiring deployment on heroku
 - More robust set of data uploaded to the songData.js file
 - Potentially expanding endpoints to more efficiently access genre table
